float64[] angles
int64[] state_foot
